utils/providers.py

```diff
@@ -307,14 +309,19 @@ def _setup_providers(prov_class, validate, check_existing):
 
     """
 
+    prov_class_map = {
```

Comment:
okay, I did look at BaseProvider.type_mapping, but I couldn't confirm that it was getting populated with the strings 'cloud', 'infra', 'middleware', 'container'. I'll test this.